pkg/build/admission/alwayspull.go

```diff
@@ -0,0 +1,58 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func init() {
+	admission.RegisterPlugin("AlwaysPullBuildImages", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return NewAlwaysPullBuildImages(), nil
+	})
+}
+
+type alwaysPull struct {
+	*admission.Handler
+}
+
+// NewAlwaysPullBuildImages returns an admission controller for builds that sets ForcePull to true
+// regardless of the user's desired value. This is to ensure that a user has access to use the
+// image, since once an image has been pulled to a given node, access to that image is not
+// restricted or verified in any way (yet).
+func NewAlwaysPullBuildImages() admission.Interface {
+	return &alwaysPull{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+func (a *alwaysPull) Admit(attr admission.Attributes) error {
+	if resource := attr.GetResource(); resource != buildsResource && resource != buildConfigsResource {
```

Comment:
That's the plan. Update coming soon.

On Friday, January 15, 2016, Cesar Wong <notifications@github.com> wrote:

> In pkg/build/admission/alwayspull.go
> <https://github.com/openshift/origin/pull/6680#discussion_r49878222>:
>
> > +type alwaysPull struct {
> > +	*admission.Handler
> > +}
> > +
> > +// NewAlwaysPullBuildImages returns an admission controller for builds that sets ForcePull to true
> > +// regardless of the user's desired value. This is to ensure that a user has access to use the
> > +// image, since once an image has been pulled to a given node, access to that image is not
> > +// restricted or verified in any way (yet).
> > +func NewAlwaysPullBuildImages() admission.Interface {
> > +	return &alwaysPull{
> > +		Handler: admission.NewHandler(admission.Create, admission.Update),
> > +	}
> > +}
> > +
> > +func (a *alwaysPull) Admit(attr admission.Attributes) error {
> > +	if resource := attr.GetResource(); resource != buildsResource && resource != buildConfigsResource {
>
> The details subresource is meant to only update the build's spec.Source
> and likely should be ignored. Actually I think it would be sufficient to
> only check on the builds resource with an empty subresource. That way
> you're also guaranteed to only get Build objects.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6680/files#r49878222>.
>